test/adapter/json_api/relationship_test.rb

```diff
@@ -0,0 +1,173 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class RelationshipTest < ActiveSupport::TestCase
+          RelationshipAuthor = Class.new(::Model)
+          class RelationshipAuthorSerializer < ActiveModel::Serializer
+            has_one :bio do
+              link :self, '//example.com/link_author/relationships/bio'
+            end
+
+            has_one :profile do
+              link :related do
+                "//example.com/profiles/#{object.profile.id}"
+              end
+            end
+
+            has_many :locations do
+              link :related do
+                ids = object.locations.map!(&:id).join(',')
+                href "//example.com/locations/#{ids}"
+              end
+            end
+
+            has_many :posts do
+              link :related do
+                ids = object.posts.map!(&:id).join(',')
+                href "//example.com/posts/#{ids}"
```

Comment:
I also figured out that there is a bug here. Using:

```ruby
has_many :posts do
  link :related do
    ids = object.posts.map!(&:id).join(',')
    meta ids: ids
  end
end
# => { links: { posts: { related: { href: nil, meta: { ids: [1337] } }  } } }
```
results in a `nil` href. According to JSONAPI, we should be able to use `meta` without `href` (http://jsonapi.org/format/#document-links).